app/src/forms/form/service.js

```diff
@@ -416,6 +416,58 @@ const service = {
       .withGraphFetched('statusCode')
       .where('formId', formId);
   },
+
+  // -----------------------------------------------------------------------------
+  // API Key
+  // -----------------------------------------------------------------------------
+  // Modification actions are audited in form_api_key_audit via a trigger
+
+  // Get the current key for a form
+  readApiKey: async (formId) => {
+    return FormApiKey.query()
+      .modify('filterFormId', formId)
+      .first();
+  },
+
+  // Add an API key to the form, delete any existing key
+  createOrReplaceApiKey: async (formId, currentUser) => {
+    let trx;
+    try {
+      const currentKey = await service.readApiKey(formId);
+      trx = await transaction.start(FormApiKey.knex());
+
+      const newKey = {
+        formId: formId,
+        secret: uuidv4(),
+        createdBy: currentUser.username
+      };
+
+      // delete existing key for the form.
+      if (currentKey) {
+        await FormApiKey.query(trx).deleteById(currentKey.id);
+      }
+
+      // add a record using this schema, mark as published and increment the version number
+      await FormApiKey.query(trx).insert(newKey);
+
+      await trx.commit();
+
+      // return the new key
+      return await service.readApiKey(formId);
```

Comment:
Don't think we need an await on this - function should be returning a Promise instead of the resultant value.